src/control/server/storage/scm/provider.go

```diff
@@ -562,9 +565,18 @@ func (p *Provider) CheckFormat(req FormatRequest) (*FormatResponse, error) {
 		}
 
 		p.log.Debugf("device %s filesystem: %s", req.Dcpm.Device, fsType)
-		if fsType != fsTypeNone {
-			return res, nil
+
+		switch fsType {
+		case fsTypeExt4:
+			res.Mountable = true
+		case fsTypeNone:
+			res.Formatted = false
 		}
+
+		return res, nil
+	} else {
```

Comment:
in order to remove this else statement (as per solid principal advice) should flip this and check for "if req.Dcpm == nil" returning early with res.Mountable == true then proceed with dcpm logic reducing indentation and preserving leftmost happy path.

Reply:
Done.